### PR TITLE
Feature: initial version of the Social Presence Plugin for Steam

### PR DESCRIPTION
**Pull request by TrueBrain**

### CODE REVIEW

File: src/plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#ifdef _WIN32
+#   define WIN32_LEAN_AND_MEAN
+#   include <windows.h>
+#   define DLL_EXPORT __declspec(dllexport)
+#else
+#   define DLL_EXPORT
+#endif /* WIN32 */
+
+#include <chrono>
+#include <time.h>
```
Comment (rubidium42) [inline src/plugin.cpp]: What's using chrono and time?

**Review by rubidium42 — COMMENTED** [review]
The icon looks an awful lot like the icon of OpenTTD, so arguably it's GPLv2 licensed and putting it here under another license might be troublesome. How needed is the icon for a library (under MacOS)?

**Comment by TrueBrain** [comment]
> The icon looks an awful lot like the icon of OpenTTD, so arguably it's GPLv2 licensed and putting it here under another license might be troublesome. How needed is the icon for a library (under MacOS)?

As I can't remember who has drawn them, good point. I will remove them.

Without them, it looks a bit odd on MacOS. But if it annoys someone enough, they will give us MIT licensed icons :)

Good point, tnx for being this sharp :D

(PS: I will apply this on the other 2 repos as well ofc)

**Review by rubidium42 — APPROVED** [review]
The C++ code looks okay to me. Can't say much useful about the cmake and workflows.

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**